tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -247,22 +285,139 @@ def emit_view_body(fn: NativeFunctionWithDifferentiabilityInfo, var: str) -> Tup
         # See NOTE [ View + Inplace detection ] for more details about this logic
         if is_tensor_list_type(return_info.type):
             if base_name in MULTI_OUTPUT_SAFE_FUNCTIONS:
-                creation_meta = 'CreationMeta::MULTI_OUTPUT_SAFE'
+                creation_meta = 'torch::autograd::CreationMeta::MULTI_OUTPUT_SAFE'
             else:
-                creation_meta = 'CreationMeta::MULTI_OUTPUT_NODE'
-            call += (f'as_view(/* base */ {view_info}, /* output */ {var}, /* is_bw_differentiable */ true, '
+                creation_meta = 'torch::autograd::CreationMeta::MULTI_OUTPUT_NODE'
+            call += (f'torch::autograd::as_view(/* base */ {view_info}, /* output */ {var}, /* is_bw_differentiable */ true, '
                      '/* is_fw_differentiable */ true, '
                      f'/* creation_meta */ {creation_meta});')
             rhs_value = f'std::move({var})'
         else:
             _, unpacked_bindings = unpack_args(f)
             call += emit_view_lambda(f, unpacked_bindings)
-            creation_meta = 'GradMode::is_enabled() ? CreationMeta::DEFAULT: CreationMeta::NO_GRAD_MODE'
-            rhs_value = (f'as_view(/* base */ {view_info}, /* output */ {var}, /* is_bw_differentiable */ true, '
+            creation_meta = ('at::GradMode::is_enabled() ? '
+                             'torch::autograd::CreationMeta::DEFAULT : '
+                             'torch::autograd::CreationMeta::NO_GRAD_MODE')
+            rhs_value = (f'torch::autograd::as_view(/* base */ {view_info}, /* output */ {var}, /* is_bw_differentiable */ true, '
                          '/* is_fw_differentiable */ true, '
                          f'/* view_func */ func, /* creation_meta */ {creation_meta})')
     else:
         # This could be supported but we don't need it at the moment, so keeping things simple.
         raise RuntimeError('Function that return multiple differentiable output '
                            'when at least one of them is view is not supported.')
     return call, rhs_value
+
+def emit_inplace_or_view_body(fn: NativeFunctionWithDifferentiabilityInfo) -> List[str]:
+    f = fn.func
+    inplace_view_body: List[str] = []
+
+    dispatcher_sig = DispatcherSignature.from_schema(f.func)
+    dispatcher_exprs = dispatcher_sig.exprs()
+
+    ret_and_arg_types = ', '.join([dispatcher_sig.returns_type()] + [a.type.cpp_type() for a in dispatcher_exprs])
+    # code-generated tracing kernels plumb and recompute dispatch keys directly through the kernel for performance.
+    # See Note [Plumbing Keys Through The Dispatcher] for details.
+    dispatch_key_set = 'ks & c10::DispatchKeySet(c10::DispatchKeySet::FULL_AFTER, c10::DispatchKey::InplaceOrView)'
```

Comment:
The one used by VariableType is `ks & c10::after_autograd_keyset`
Why do you need something more complex here than `ks & c10::after_autograd_keyset & !c10::DispatchKey::InplaceOrView` ? Also what is the `FULL_AFTER` key?

Reply:
Ho.
That FULL_AFTER is a bit weird, since it's a keyset I was expecting it to be a key... ok